src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -320,38 +325,92 @@ const NetworkAssetSelectInput = ({
       await getSmartWalletBalancesByChain(walletAddress, supportedChains);
     };
     handleBalanceGet();
-  }, [supportedChains, sdk, walletAddress]);
+  }, [supportedChains, sdk, accountAddress]);
+
+  useEffect(() => {
+    const handleKeybasedBalanceGet = async () => {
+      if (!sdk || !providerAddress) return;
+      await getKeybasedWalletBalancesPerChain(providerAddress, supportedChains);
+    };
+    handleKeybasedBalanceGet();
+  }, [supportedChains, sdk, providerAddress]);
 
   const filteredSelectedNetworkAssets = useMemo(() => {
     const filtered = selectedNetworkAssets.filter((asset) => containsText(asset.name, assetSearchQuery) || containsText(asset.symbol, assetSearchQuery));
     return orderBy(filtered, [(asset) => asset.balanceWorthUsd ?? 0, 'name'], ['desc', 'asc']);
   }, [selectedNetworkAssets, assetSearchQuery]);
 
   useEffect(() => {
-    const updateAvalancheBalance = () => {
+    const updateAvalancheBalanceSmart = () => {
       if (!sdk ||
         !walletAddress ||
         !preselectedNetwork ||
         preselectedNetwork.chainId !== CHAIN_ID.AVALANCHE ||
         !smartWalletBalanceByChain ||
-        filteredSelectedNetworkAssets.length
+        !filteredSelectedNetworkAssets.length ||
+        isLoadingAssets
       ) {
         return;
       }
       setsmartWalletBalanceByChain([
-        ...smartWalletBalanceByChain,
+        ...smartWalletBalanceByChain.filter(
+          (element) => element.chain !== CHAIN_ID.AVALANCHE
+        ),
         {
           total: sumAssetsBalanceWorth(filteredSelectedNetworkAssets),
-          title: (supportedChains.find(({ chainId }) =>  chainId === CHAIN_ID.AVALANCHE) as Chain).title,
+          title: (
+            supportedChains.find(
+              ({ chainId }) => chainId === CHAIN_ID.AVALANCHE
+            ) as Chain
+          ).title,
           chain: CHAIN_ID.AVALANCHE,
         },
       ]);
     };
-    updateAvalancheBalance();
+    updateAvalancheBalanceSmart();
   }, [
     sdk,
     supportedChains,
-    walletAddress,
+    accountAddress,
+    preselectedNetwork,
+    filteredSelectedNetworkAssets,
+    isLoadingAssets,
+  ]);
+
+  useEffect(() => {
+    const updateAvalancheBalanceKeybased = () => {
```

Comment:
Same as for other smart wallet / key based balance methods: maybe we can move all and combine? some code can definitely be reused and every time we would need total balances we now would need to move this code into other components which would be wrong

Reply:
This way we would also be able to update the balance using hook (if method exposed)